docs/contributing/technical/resolve-merge-conflicts.md

```diff
@@ -6,138 +6,79 @@ keywords:
+**NOTE**: If you have already opened up a pull request, then you can see if there is a conflict at the bottom of the PR. Even though it is possible to resolve conflicts through GitHub, it is best practice to resolve them locally.
 
-```shell
-git diff package.json
-```
-
-It will look like this:
+![merge conflicts git message](../../../static/img/contributing-resolve-merge-conflicts-dont-do.png)
 
-![review merge conflicts](../../../static/img/contributing-resolve-merge-conflicts-review-conflicts.png)
+## How to Resolve Merge Conflicts
 
-## Resolve Conflicts
+If you have merge conflicts, it is best to use a text editor to resolve them. Identify the conflicting files and open them up in your editor of choice. For the conflicting sections, you have an option to either keep the incoming changes, keep your changes or keep both sets of changes. The conflicting sections will be marked with `<<<<<<<`, `=======`, and `>>>>>>>` symbols.
 
-Since this pull request does not modify the `package.json` file, it is safe to fast-forward the changes from `origin/main`:
-
-```shell
-# overwrite with origin/main changes
-git show :3:package.json > package.json
-```
+Here is a [guide](https://dev.to/adiatiayu/how-to-resolve-merge-conflicts-using-the-merge-editor-feature-on-vs-code-pic) for an in depth walkthrough of the process. Once you have resolved all of the conflicts, then you can stage, commit and push your changes to your branch.
 
-A more traditional way of doing the same thing is:
-
-```shell
-# make a local copy of all changes and use --theirs
-# --theirs strategy overwrite with origin/main changes
-git show :1:package.json > base.package.json
-git show :2:package.json > branch.package.json
-git show :3:package.json > head.package.json
-git merge-file -p --theirs \
-    branch.package.json base.package.json head.package.json > package.json
+```bash
+git status
+git add .
+git commit -m "fix: resolving merge conflicts"
+git push
 ```
 
-## Commit Changes
+## How to Keep Your Branch Updated
 
-Not making any assumptions about editor preferences, running this will open the configured editor with a default commit message:
+It is common for your branch to fall behind the main repository's branch. So it is important to keep it up to date as you are contributing.
 
-```shell
-git commit
-```
+### Using GitHub
 
-That should look like this:
+To update your branch on GitHub, you can go to your forked copy of the project and click on `Sync fork` and then the `Update branch` button.
 
-![commit merge message](../../../static/img/contributing-resolve-merge-conflicts-commit-message.png)
+![syncing your branch on GitHub](../../../static/img/sync-branch-GH.png)
 
-## Push Changes
+### Using Git
 
-One more security check to make sure your branch has not diverged and push:
+To update your branch using Git and the terminal, you can use the following command:
 
-```shell
-git status
-git push
+```bash
+git pull upstream <main-branch-name>
```

Comment:
I don't think we can we pull changes from `upstream` when we haven't add the remote locally. Or maybe I'm wrong?

So, we can do either:

- Add an instruction to add `upstream` locally.
OR
- Because a screenshot for updating `origin` repo is provided here, and that folks will automatically have `origin` after they clone the repo locally, we might want to change this to `origin`. This also applies to the rest of the suggestions if we choose this option.

```suggestion
git pull origin <main-branch-name>
```

Reply:
> I don't think we can we pull changes from upstream when we haven't add the remote locally. But I could be wrong.

I had assumed they already did that before reaching this guide. But we can add back in the instructions for adding the upstream locally 👍